test/filters/test_crop_hull.cpp

```diff
@@ -0,0 +1,479 @@
+            p.getVector3fMap() += baseOffset.getVector3fMap();
+            return p;
+          };
+          pcl::CropHull<pcl::PointXYZ> cropHullFilter = createDefaultCropHull(inputCloud);
+          std::vector<TestData> testDataSuite = createTestDataSuite(insidePointGenerator, outsidePointGenerator);
+          for (TestData const & testData : testDataSuite)
+          {
+            cropHullFilter.setInputCloud(testData.inputCloud);
+            checker(cropHullFilter, testData);
+          }
+        }
+      }
+
+    private:
+      pcl::CropHull<pcl::PointXYZ> createDefaultCropHull (pcl::PointCloud<pcl::PointXYZ>::ConstPtr inputCloud) const
+      {
+        pcl::CropHull<pcl::PointXYZ> cropHullFilter(true);
+        pcl::ConvexHull<pcl::PointXYZ> convexHull;
+        convexHull.setDimension(3);
+        convexHull.setInputCloud(inputCloud);
+        pcl::PointCloud<pcl::PointXYZ>::Ptr hullCloudPtr(new pcl::PointCloud<pcl::PointXYZ>);
+        std::vector<pcl::Vertices> hullPolygons;
+        convexHull.reconstruct(*hullCloudPtr, hullPolygons);
+        cropHullFilter.setHullIndices(hullPolygons);
+        cropHullFilter.setHullCloud(hullCloudPtr);
+        cropHullFilter.setDim(getDim());
+        return cropHullFilter;
+      }
+
+      mutable std::mt19937 gen;
+      mutable std::uniform_real_distribution<float> rd;
+      pcl::PointCloud<pcl::PointXYZ> baseOffsetList;
+  };
+
+
+  class PCLCropHullTestFixture2d : public PCLCropHullTestFixtureBase
+  {
+    public:
+      PCLCropHullTestFixture2d()
+        : baseInputCloud(new pcl::PointCloud<pcl::PointXYZ>)
+      {
+        baseInputCloud->emplace_back(0.0f, 0.0f, 0.0f);
+        baseInputCloud->emplace_back(0.0f, 1.0f, 0.0f);
+        baseInputCloud->emplace_back(1.0f, 0.0f, 0.0f);
+        baseInputCloud->emplace_back(1.0f, 1.0f, 0.0f);
+        baseInputCloud->emplace_back(0.0f, 0.0f, -0.1f);
+        baseInputCloud->emplace_back(0.0f, 1.0f, -0.1f);
+        baseInputCloud->emplace_back(1.0f, 0.0f, -0.1f);
+        baseInputCloud->emplace_back(1.0f, 1.0f, -0.1f);
+      }
+    protected:
+      pcl::PointCloud<pcl::PointXYZ>::ConstPtr getBaseInputCloud() const override {
+        return baseInputCloud;
+      }
+      int getDim() const override {
+        return 2;
+      }
+    private:
+      pcl::PointCloud<pcl::PointXYZ>::Ptr baseInputCloud;
+  };
+
+
+  class PCLCropHullTestFixture3d : public PCLCropHullTestFixtureBase
+  {
+    public:
+      PCLCropHullTestFixture3d()
+        : baseInputCloud(new pcl::PointCloud<pcl::PointXYZ>)
+    {
+      baseInputCloud->emplace_back(0.0f, 0.0f, 0.0f);
+      baseInputCloud->emplace_back(0.0f, 0.0f, 1.0f);
+      baseInputCloud->emplace_back(0.0f, 1.0f, 0.0f);
+      baseInputCloud->emplace_back(0.0f, 1.0f, 1.0f);
+      baseInputCloud->emplace_back(1.0f, 0.0f, 0.0f);
+      baseInputCloud->emplace_back(1.0f, 0.0f, 1.0f);
+      baseInputCloud->emplace_back(1.0f, 1.0f, 0.0f);
+      baseInputCloud->emplace_back(1.0f, 1.0f, 1.0f);
```

Comment:
Any reason for the particular data in base input cloud? Please add a note regarding the size and data.

```suggestion
      for (const i: {0.f, 1.f})
        for (const j: {0.f, 1.f})
          for (const k: {0.f, 1.f})
            baseInputCloud->emplace_back(i, j, k);
```